### DISCUSSION TITLE
Should write_tasks_per_group be the same as LAYOUT_X or LAYOUT_Y in predef_grid_params.yaml?

### DISCUSSION THREAD

**Discussion by lo-y-wni** [ufs-community/ufs-srweather-app#724] [Q&A] [answered]
Hello everyone:

I have a question about IO quilting option. I know the total processors = LAYOUT_X * LAYOUT_Y + write_tasks_per_group * write_groups. Suppose write_groups is 1, I want to use **total 144 processors** and hope **LAYOUT_X=14** and **LAYOUT_Y=10**, then **write_tasks_per_group must be 4** at this configuration. But I don't know is the configuration illegal or not since the predefined domain are almost let write_tasks_per_group=LAYOUT_Y or write_tasks_per_group=LAYOUT_X.

Yi-Chuan

**Comment by gspetro-NOAA** [comment]
Hi Yi-Chuan, 

I've reached out to @EdwardSnyder-NOAA, and he is looking into this for you. He should get back to you within a few days. 

Best,
Gillian Petro, Advanced User Support (AUS) Team

**Comment by EdwardSnyder-NOAA** [comment, marked as answer]
Good afternoon Yi-Chuan,

The configuration you described isn't illegal. The `LAYOUT_X` and `LAYOUT_Y` variables (number of MPI tasks to use in the horizontal x and y directions) are independent from the `write_tasks_per_group` variable (number of MPI tasks to allocate for each write group). I'm not sure why the predefined grids have some of these variables equal to each other. But you should be able to run the model with this configuration. 

Please let me know if you have any additional questions! And I'll address your other question soon!

Thanks!

Eddie Snyder | Advanced User Support (AUS) Team

**Reply by lo-y-wni** [reply]
Hi Edward:

Thanks your response! It's very happy to hear `write_tasks_per_group` doesn't need to be the same as `LAYOUT_X` or `LAYOUT_Y`. It lets me tune grid decomposition more flexible!

Yi-Chuan
